CODING_GUIDELINES.md

```diff
@@ -6,30 +6,82 @@ This document contains the coding guidelines for the ASPIRE Python project.
+scheme, tag it similarly, and complete a merge to master following the controls below. It is at this point the
+release would be a candadite to upload to PyPI.
+
+###  Git Workflow Controls
+
+To facilitate effective integration, merges into `develop` must occur as a pull request that is approved by two developers.
+One devloper can be anyone, but should probably be someone related to the effort if that is applicable.
+The second developer is a designated codeowner described in the `CODEOWNERS` file.
+
+Additionally, all outstanding requested changes and comments must be resolved before merging.  Pushing new changes should trigger
+a re-review process of changed files to prevent oversights.
+
+Code should be passing integration and unit tests as defined in code on all supported plaforms by the evolving continuous integration systems,
+or be explicitly exempted by a codeowner. This is both for code quality and so maintenence of CI systems is not overlooked.
+
+All of these controls will be managed automatically by the Gitlab server.
+
+(discussion note, maybe want an additional control (Amit?) on master/releases, I don't expect too often..).
+
+### Git Workflow Courtesy
+
+Contributors are responsible for keeping their working repositories up to date with the main Github repo.
+This is particularly important during active development.
+
+If a piece of work is known to change files that will effect other developers' active work, they should be invited to the review or tagged,
+so that the most effective order of integration can be established.  In the case of your own works having integration
+concerns, again, inviting other developers to discuss might be helpful.
+
+In the case a codeowner will not be available, such as vacation/travel, and an emergency patch can not be delayed
+a simple explicit consensus among the remaining developers should suffice.
+(discussion note, This level of urgency seems unlikely, can maybe drop and agree to delay, more likely a codeowner will just be busy,
+which is not as simple).
+
+Do not merge over another persons working branch or pull request unless explicitly permitted to do so by the author.
+Pull requests are welcomed.
+
+We will for a time attempt using Project boards in conjunction with Issues to document tasks in various stages.
+Generally this will look like a board collecting related tasks which representing a conceptual project or
+actual milestone that comes out of our meetings.  For example, an event, release, or large feature set
+with many components all are reasonable project boards.  This is for planning purposes and to facilitate coordination
+in the remote environment. If it helps we can keep doing it so long as the time and effort is accounted for.
+
+## Documentation
+
+(Todo, get an idea what exactly is and is not setup wrt gh-pages/rtd etc and discuss, also consider integration with other tools/CI)
+
+## Best Practices Reading
```

Comment:
Probably we should move this up so beginners have more chance to take a look.

Reply:
I feel its not related to a guideline or procedure at all and doesn't need to be in the document, so moving it to the bottom instead of deleting was sort of a concession, because I agree it was worth reading. To me there are clearly more important things that need to be spelled out in this guide, mainly for development, and maybe we should focus on that... Regardless, if having this at the TOF is important to you I'll move it back. My expectation is that a beginner collaborating on this project could be expected to read the whole 87 lines :).

Reply:
Moved back to TOF